projects/packages/analyzer/src/class-utils.php

```diff
@@ -31,24 +31,37 @@ static function get_param_default_as_string( $default, $current_class ) {
 		}
 	}
 
-	static function node_to_class_name( $node, $class_for_self = null ) {
+	/**
+	 * Gets string representation of the Node's class
+	 *
+	 * @param Node $node Node.
+	 * @param any  $class_for_self Class reference.
+	 * @return string
+	 */
+	public static function node_to_class_name( $node, $class_for_self = null ) {
 		if ( $node instanceof Node\Expr\Variable
 			|| $node instanceof Node\Stmt\Class_ ) {
 			$class_name = $node->name;
 		} elseif ( $node instanceof Node\Name ) {
 			$class_name = '\\' . implode( '\\', $node->parts );
 		} elseif ( $node instanceof Node\Expr\PropertyFetch ) {
 			$class_name =
-						'$'
-						. self::maybe_stringify( $node->var->name )
-						. '->' . self::maybe_stringify( $node->name->name );
+				'$'
+				. self::maybe_stringify( $node->var->name )
+				. '->' . self::maybe_stringify( $node->name->name );
 		} elseif ( $node instanceof Node\Expr\ArrayDimFetch ) {
 
-			$class_name =
-						'$'
-						. self::maybe_stringify( $node->var->name )
-						. '[' . self::maybe_stringify( $node->dim->value )
-						. ']';
+			$dim_val = '';
+			if ( $node->dim instanceof Node\Expr\Variable ) {
+				$dim_val = '[$' . self::maybe_stringify( $node->dim->name ) . ']';
+			} elseif ( $node->dim instanceof Node\Expr\PropertyFetch ) {
+				$dim_val = '[$' . self::maybe_stringify( $node->dim->var->name ) . ']';
+			} elseif ( $node->dim instanceof Node\Expr\ArrayDimFetch ) {
+				$dim_val = '[$' . self::maybe_stringify( $node->dim->var->name ) . '["' . self::maybe_stringify( $node->dim->dim->value ) . '"]]';
+			} else {
+				$dim_val = '[' . self::maybe_stringify( $node->dim->value ) . ']';
```

Comment:
Should the output string here be quoted, like you're doing in the `getArrayToken()` being added in the other file?

Reply:
Good call. But I'm not sure tbh :D Quotes were omitted in the initial implementation (line 50 before my changes). I need to dig deeper into the `Differences` APIs to better understand how things work. But I'd say the quoting is out of scope of this PR